code/c++/search_letor.cc

```diff
@@ -0,0 +1,94 @@
+#include <xapian-letor.h>
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+using namespace std;
+
+static void show_usage()
+{
+    cout << "Usage: rank_letor --db=DIRECTORY MODEL_METADATA_KEY QUERY\n";
```

Comment:
*  `rank_letor` doesn't match `search_letor.cc`
* The code actually expects just a database path for the first parameter, not with `--db=`
* `DIRECTORY` is inaccurate - a Xapian database can be a single file (glass supports a read-only single file variant, and there's also stub database files).